docs/USAGE_GUIDE.md

```diff
@@ -0,0 +1,529 @@
+        getPasswordInput().clearAndSendKeys(password);
+        clickLoginButton();
+    }
+}
+```
+
+Let's break down what we see in the example above.
+
+1. `BaseComponent` Extension on `HeaderComponent`
+   1. All Components should inherit functionality from the `BaseComponent` class. Similar to Page Objects, there are some extremely helpful methods there that allow access the same access as mentioned from Page Objects, but also the construction of Component lists. That's covered later.
+2. Properties (or aka Java Fields)
+   1. Exactly the same fundamentals as the Page Object. These fields are the elements that exist on the page. With a "non-list" component, we always want to provide the fully qualified By locator that includes the parent.
+3. Getters
+   1. Standard getter functions are auto generated with the lombok `@Getter` annotation from the class level. Any time we want to interact with an element, we
+      always call the getter instead of the raw field.
+4. No Constructor
+   1. Here, we don't need to customize our constructor with a verification method. The Page Object is responsible for this.
+5. Helper Functions
+   1. We can include any additional helper functions here to make our calling code easier to read.
+6. Instantiating the Component on the Page Object
+   1. The Page Object can then instantiate a new instance of the Component we just wrote. Code calling the loginPage will now have access to getHeaderComponent()'s getters and helper functions!
+
+##### Component Example - List Context
+Building a Component under a "list" context is fundamentally the same. In the search results page example mentioned from the Components section, we
+can define properties exactly the same way. The only difference is that a list Component doesn't define properties with the parent in the `By` locator.
+The search results page would then create the list of the search results with `BaseComponent`'s `buildComponentList()` method. 
+
+**The Component**
+```java
+@Getter
+public class SearchResultItem extends BaseComponent {
+
+   private final DivWebElement itemName = new DivWebElement(".inventory_item_name");
+   private final DivWebElement itemDescription = new DivWebElement(".inventory_item_desc");
+   private final DivWebElement itemPrice = new DivWebElement(".inventory_item_price");
+   private final ButtonWebElement addToCart = new ButtonWebElement(".btn_primary");
+
+   public void clickAddToCartButton() {
+      getAddToCart().click();
+   }
+}
+```
+
+**The Page Object**
+```java
+@Getter
+public class SearchResultsPage extends BasePage {
+   private final static String INVENTORY_ITEM_SELECTOR = ".inventory_item";
+   private final HeaderComponent headerComponent = new HeaderComponent();
+   private final DivWebElement inventoryListContainer = new DivWebElement(By.cssSelector(".inventory_list"));
+   private final DropDownWebElement sortDropDown = new DropDownWebElement(".product_sort_container");
```

Comment:
The constructor `BaseWebElement(String cssSelector)` is like a helper constructor in that it cuts down on code if you're wanting to just use a css selectors across the board. Otherwise, you'd have to pass in a `By` locator, `BaseWebElement(By by)`.

It just shaves off a little bit of code.
`private final DivWebElement element = new DivWebElement("element_locator");`
vs
`private final DivWebElement element = new DivWebElement(By.cssSelector("element_locator));`